sktwocnt/lib/bisection.f90

```diff
@@ -1,36 +1,49 @@
-!> Contains routines to locate a value in an array using bisection.
+!> Module that contains routines to locate a value in an array using bisection.
 module bisection
 
-  use common_accuracy, only: dp
+  use common_accuracy, only : dp
 
   implicit none
   private
 
   public :: bisect
 
-  !> Bisection driver.
+  !> Bisection driver that interfaces integer- and real-valued array routines.
   interface bisect
     module procedure bisect_real
     module procedure bisect_int
   end interface bisect
 
+
 contains
 
-  !> Real case for bisection search to to find a point in an array xx(:)
-  !! between xx(1) and xx(size(xx)) such that element indexed ind is less than
-  !! the value x0 queried.
-  !! \param xx Array of values in monotonic order to search through.
-  !! \param x0 Value to locate ind for.
-  !! \param ind Located element such that xx(ind) < x < xx(ind).
+  !> Real case for bisection search to to find a point in an array xx(:) between xx(1) and
+  !! xx(size(xx)) such that element indexed ind is less than the value x0 queried.
   pure subroutine bisect_real(xx, x0, ind, tol)
-    real(dp), intent(in) :: xx(:), x0
+
+    !> array of values in monotonic order to search through
+    real(dp), intent(in) :: xx(:)
+
+    !> value to locate ind for
+    real(dp), intent(in) :: x0
+
+    !> located element such that xx(ind) < x0 < xx(ind)
     integer, intent(out) :: ind
+
+    !> optional, user-specified tolerance for comparisons
     real(dp), intent(in), optional :: tol
 
+    !> length of array to search
```

Comment:
Don't add docmarkers for variables not part of the function signature. Ford will lump them together with the some of the other docstrings and create incorrect API docs.

Reply:
I see, I wasn't aware of that. I can change the docmarkers to ordinary comments.

Reply:
What about declarations on the module level (before 'contains'), like some of the tolerance/length specifications ('!!' or '!>')?

Reply:
Those variables are `private` / `protected` and are captured by Ford with their docstring, but not necessarily displayed unless enabled in the project file.

Reply:
Alright, then the current state is fine I guess.